### ISSUE TITLE
Set JDT CodeMining as Experimental

### ISSUE THREAD

**Issue by angelozerr** [angelozerr/jdt-codemining#66]
I have the intention to contribute JDT CodeMining step by step. I will start with https://bugs.eclipse.org/bugs/show_bug.cgi?id=529127 to contribute at first with references/implementations code minings.

To avoid having conflict with JDT, I have decided to mark JDT CodeMining as experimental:

 * preferences names will be names with `java.codemining.experimental*.`
 * the preference menu will be renamed to `Code Minings (Experimental)`

In other words when https://bugs.eclipse.org/bugs/show_bug.cgi?id=529127 will be finished, we will have 2 entries in the preferences:

![image](https://user-images.githubusercontent.com/1932211/43457580-a3ff9814-94c7-11e8-8f1b-468d414e2232.png)

 * Code Minings -> will come from JDT
 * Code Minings (Experimental) -> will come from JDT CodeMining

**Comment by angelozerr** [comment]
@vogella @mickaelistria, I ping you about this issue to know my plan.

**Comment by mickaelistria** [comment]
That seems good to me!
